nsqd/topic.go

```diff
@@ -275,6 +275,7 @@ func (t *Topic) messagePump() {
 				chanMsg.deferred = msg.deferred
 			}
 			if chanMsg.deferred != 0 {
+				atomic.AddUint64(&channel.messageCount, 1)
```

Comment:
Ahhh, I missed that this was an increment for the _channel_ message count. Yes, it looks like you are correct. However, it doesn't make sense that the topic is now responsible for atomically incrementing this channel's counter, we should push this responsibility into the channel logic somewhere.